Source/Core/Core/PowerPC/JitArm64/JitArm64_FloatingPoint.cpp

```diff
@@ -261,12 +261,18 @@ void JitArm64::frspx(UGeckoInstruction inst)
   const bool single = fpr.IsSingle(b, true);
   if (single)
   {
-    // Source is already in single precision, so no need to do anything but to copy to PSR1.
     const ARM64Reg VB = fpr.R(b, RegType::LowerPairSingle);
     const ARM64Reg VD = fpr.RW(d, RegType::DuplicatedSingle);
 
-    if (b != d)
-      m_float_emit.FMOV(EncodeRegToSingle(VD), EncodeRegToSingle(VB));
+    const ARM64Reg V0 = fpr.GetReg();
+
+    // No conversion is needed in this case, but we still need to handle NaNs and flush denormals
+    // (if enabled in FPSCR) like almost any float operation does. We accomplish this by adding 0.0,
+    // which should be cheaper than FCVT 32 -> 64 followed by FCVT 64 -> 32.
+    m_float_emit.MOVI(8, EncodeRegToSingle(V0), 0);
+    m_float_emit.FADD(EncodeRegToSingle(VD), EncodeRegToSingle(VB), EncodeRegToSingle(V0));
```

Comment:
This will turn -0.0 into 0.0. To avoid this, add -0.0.

Reply:
Thanks, I was not aware of that :)

Reply:
Should be fixed now, assuming I've used `MOVI` correctly.

Reply:
Exception if FPSCR.RMode is TowardsMinusInfinity, in which case you'd want to add +0.0 to prevent +0.0 from being converted to -0.0.

Reply:
Hmm... Maybe I should just go with the double FCVT then. Having to re-emit code when the rounding mode changes sounds annoying. This case hopefully won't be triggered often anyway.

Reply:
Briefly grepping the codebase apparently we *don't* currently adjust FPCR based on guest fpscr, which is slightly surprising.

Reply:
That's what the first commit of this PR fixes.

Reply:
Okay, I've pushed a new approach (as the last commit of this PR this time). It simply adds an additional check to the FMOV path so that we only use it if the register is store-safe.